src/qibo/models/circuit.py

```diff
@@ -120,7 +120,7 @@ class Circuit:
     A specific backend has to be used for performing calculations.
 
     Args:
-        nqubits (int): Total number of qubits in the circuit.
+        nqubits (int, list): Number of qubits in the circuit or a list of wire names.
```

Comment:
It could also be `None`. Though I'm not sure I want to advertise it...

This argument listing is unsuitable to multiple signatures, the free form would have better served the purpose. Maybe, the best thing we could is providing a couple of examples on top (with `nqubits` and with `wire_names`).

Reply:
[a4f2092](https://github.com/qiboteam/qibo/pull/1500/commits/a4f2092708a9f15e066e28fdeb6da18eaaaa302f)

Please review the updated docstring, and if everything ok, please close this conversation.